### PR TITLE
Add support for versioned toolchains

### PR DESCRIPTION
**Pull request by stellar-aria**
Add support for multiple simultaneous toolchain installations, for example for release and upstream builds using different versions. This will need to be merged into _both_ release and community in order to work properly.

This places arch-specific toolchain installs underneath a folder corresponding to the version.

For ex.
```
toolchain/
    - v11
        - linux-x86_64/
    - v10/
        - linux-x86_64/
```

### CODE REVIEW

File: scripts/toolchain/dbtenv.sh [hunk 1]
```diff
@@ -244,7 +238,7 @@ dbtenv_cleanup()
     
     # Kludgey but, quick path to getting rid of these
     printf "Removing any nuisance mac dotfiles..";
-    find toolchain | grep -e '\/[.][^\/]*$' | sed -e 's/\(.*\)/rm \"\1\"/' | /usr/bin/env bash;
+    find toolchain | grep -e '/[.][^\/]*$' | sed -e 's/\(.*\)/rm \"\1\"/' | /usr/bin/env bash;
```
Comment (m-m-adams) [inline scripts/toolchain/dbtenv.sh:241]: why is the escape removed?

File: scripts/toolchain/dbtenv.sh [hunk 2]
```diff
@@ -153,7 +153,7 @@ dbtenv_get_kernel_type()
     if [[ "${SYS_TYPE}" == "darwin" || "${SYS_TYPE}" == "linux" ]]; then
         # Disabling rosetta checking now that we've got native arm64
         # dbtenv_check_rosetta || return 1;
-        TOOLCHAIN_ARCH_DIR="${DBT_TOOLCHAIN_PATH}/toolchain/${SYS_TYPE}-${ARCH_TYPE}";
+        TOOLCHAIN_ARCH_DIR="${DBT_TOOLCHAIN_PATH}/toolchain/v${DBT_TOOLCHAIN_VERSION}/${SYS_TYPE}-${ARCH_TYPE}";
         TOOLCHAIN_URL="https://github.com/litui/dbt-toolchain/releases/download/v${DBT_TOOLCHAIN_VERSION}/dbt-toolchain-${DBT_TOOLCHAIN_VERSION}-${SYS_TYPE}-${ARCH_TYPE}.tar.gz";
```
Comment (m-m-adams) [inline scripts/toolchain/dbtenv.sh:157]: update URL here too

File: scripts/toolchain/dbtenv.sh (hunk 1 above)
Comment (stellar-aria) [reply scripts/toolchain/dbtenv.sh:241]: it was causing an error message on linux

File: scripts/toolchain/dbtenv.sh (hunk 2 above)
Comment (stellar-aria) [reply scripts/toolchain/dbtenv.sh:157]: Fixed

File: scripts/cmake/CMakeToolchainDeluge.cmake [hunk 3]
```diff
@@ -7,6 +7,9 @@ else()
   set(TOOLCHAIN_ROOT ".")
 endif()
 
+file(READ ${TOOLCHAIN_ROOT}/toolchain/REQUIRED_VERSION TOOLCHAIN_VERSION)
```
[the remark this answers is not in the document]
Comment (stellar-aria) [reply scripts/cmake/CMakeToolchainDeluge.cmake]: Should be fixed now

**Comment by bobtwinkles** [comment]
With https://github.com/stellar-aria/dbt-toolchain/pull/2 and then https://github.com/SynthstromAudible/dbt-toolchain/pull/3 merged, this should work

**Comment by stellar-aria** [comment]
Currently just waiting on https://github.com/SynthstromAudible/dbt-toolchain/pull/3

**Pull request merged**